utilities/dataflow_kafka_to_hcapi/dataflow_kafka_to_hcapi.py

```diff
@@ -0,0 +1,529 @@
+    KafkaConsumerConfig['fetch.max.wait.ms']="500"
+    KafkaConsumerConfig['request.timeout.ms']="120000"
+
+    # Update Kafka consumer config to the following if connecting to confluent Kafka Cloud via api key and api secret
+    # KafkaConsumerConfig['sasl.mechanism']='PLAIN'
+    # KafkaConsumerConfig['security.protocol']='SASL_SSL'
+    # KafkaConsumerConfig['sasl.jaas.config']=f"""org.apache.kafka.common.security.plain.PlainLoginModule required username=\"{apiKey}\" password=\"{apiSecret}\";"""
+
+    with beam.Pipeline(options=beam_pipeline_options) as pipeline:
+        consumeRawMessage = (
+            pipeline
+            | "ReadFromKafka" >> ReadFromKafka(
+                consumer_config=KafkaConsumerConfig,
+                topics=topic_list,
+                key_deserializer = 'org.apache.kafka.common.serialization.ByteArrayDeserializer' , 
+                value_deserializer = 'org.apache.kafka.common.serialization.ByteArrayDeserializer'
+
+                # Enable the option to run pipeline in batch mode as per schedule or adhoc basis
+                # start_read_time = int((datetime.now()- timedelta(days=offset_days)).strftime("%s")) * 1000
+                )
+            | "FilterConvertKafkaRecord" >> beam.ParDo(filterKafkaRecord()).with_outputs()
+        )
+        writeFailRawMsgToGCS = (
+            consumeRawMessage.failRecord
+            | "WriteFailedConvertMsgToGCS" >> beam.ParDo(writeToGCS(),topic="fail_filter_convert_messages",bucket=gcs_archive_bucket,messageType="error")
+        )
+
+        # We can tag individual type of FHIR resource and archive it parallely from load balancing perspective and to make efficient use of resources. 
+        archivePatientResource = (
+            consumeRawMessage.patientResource
+            | "archiveMemberDetailsToGCS" >> beam.ParDo(writeToGCS(),topic="member_raw_details",bucket=gcs_archive_bucket,messageType="raw")
+        )
+        # #patient FHIR Resource
+        buildPatient = (
+            consumeRawMessage.patientResource
+            | "buildPatientResource" >> beam.ParDo(buildPatientResource()).with_outputs()
+        )
+        writeFailPatientMsgToGCS = (
+            buildPatient.failRecord
+            | "writeFailedPatientMsgToGCS" >> beam.ParDo(writeToGCS(),topic="fail_patient_transform_messages",bucket=gcs_error_bucket,messageType="error")
+        )
+        postPatientMsgToHCAPI = (
+            buildPatient.passRecord 
+            | "postPatientToHCAPI" >> beam.ParDo(postToHCAPI(),resourceType="Patient").with_outputs()
+        )
+        writeFailPostPatientMsgToGCS = (
+            postPatientMsgToHCAPI.failRecord
+            | "writeFailPatientPOSTMsgToGCS" >> beam.ParDo(writeToGCS(),topic="fail_hcapi_post_messages",bucket=gcs_error_bucket,messageType="error")
+        )
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.DEBUG, format='%(asctime)s :: %(levelname)-8s :: [%(filename)s:%(lineno)d] :: %(message)s')
```

Comment:
can we drop this log statement when publishing?

Reply:
We can but it will be useful to specify custom logging for each function or mention any warnings like logging.warning, logging.debug something like that along with the line number.